lib/realtime/helpers.ex

```diff
@@ -0,0 +1,46 @@
+defmodule Realtime.Helpers do
+  @doc """
+      iex> Realtime.Helpers.find_and_previous([1,2,3,4], fn x -> x > 2 end)
+      {2,3}
+
+      iex> Realtime.Helpers.find_and_previous([], fn x -> x > 2 end)
+      nil
+
+      iex> Realtime.Helpers.find_and_previous([3,4], fn x -> x > 2 end)
+      nil
+
+      iex> Realtime.Helpers.find_and_previous([1,2], fn x -> x > 2 end)
+      nil
+  """
+  @spec find_and_previous([element], (element -> boolean())) :: {element, element} | nil
+        when element: any()
+  def find_and_previous([], _fun) do
+    nil
+  end
+
+  def find_and_previous([first | rest], fun) do
+    if fun.(first) do
+      nil
+    else
+      find_and_previous_helper(rest, fun, first)
```

Comment:
I also wonder if using `Enum.find_index` could help simplify this. Is it possible that something like this could work?

      def find_and_previous(elements, condition) do
        elements
        |> Enum.find_index(condition)
        |> take_element_and_previous(elements)
      end

      defp take_element_and_previous(nil, _elements), do: nil
      defp take_element_and_previous(0, _elements), do: nil

      defp take_element_and_previous(index, elements),
        do: Enum.slice(elements, (index - 1)..index) |> List.to_tuple()

Reply:
It depends on the order, since it gets the first matching element and its previous, but that order doesn't need to be sorted (it just happens to be in this case). 

I'll add a description of what it does to the `@doc`.

---

I was aiming for doing it in one pass, but that's probably not actually a valuable goal. I'll do that simplification.